src/mongo_logger.ts

```diff
@@ -294,7 +308,7 @@ export interface LogConvertible extends Record<string, any> {
 
 /** @internal */
 export function stringifyWithMaxLen(value: any, maxDocumentLength: number): string {
-  const ejson = EJSON.stringify(value);
+  const ejson = EJSON.stringify(value) ?? 'custom function';
```

Comment:
Is it the application-provided custom selector? Maybe then 'custom selector' would be more descriptive rather than 'custom function'? Can we assign the same identifier multiple times or it should be unique? Do we want it to be a plain string like this or this should be some sort of stringified object format? Not sure where this being used, maybe some parsing logic later is being involved?

Reply:
Can you give an example of what `value` is for `stringifyWithMaxLen(selector, maxDocumentLength)` and `stringifyWithMaxLen(topologyDescription, maxDocumentLength)` if those are not empty?

Reply:
Yes!

The selector can take on 3 different types, I've listed them with their example `stringifyWithMaxLen` inputs below.

- Example string: `'primary'`
- Example ReadPreference object: 
```
{
"mode":"primary", "tags":null, "hedge":null, "maxStalenessSeconds":null," "minWireVersion":null
}
```
- Example Function: 
```
(err, server) => {
  if (err) {
    return this.close({ force: false }, () => exitWithError(err));
  }

  stateTransition(this, STATE_CONNECTED);
  this.emit(Topology.OPEN, this);
  this.emit(Topology.CONNECT, this);

  callback?.(undefined, this);
}
```

As for the topologyDescription, it is always a TopologyDescription object.
Here is an example: 
```
{
"type":"Single","servers":{"localhost:27017":{"address":"localhost:27017","type":"Unknown","hosts":[],"passives":[],"arbiters":[],"tags":{},"minWireVersion":0,"maxWireVersion":0,"roundTripTime":-1,"lastUpdateTime":1718729665,"lastWriteDate":0,"error":{},"topologyVersion":null,"setName":null,"setVersion":null,"electionId":null,"logicalSessionTimeoutMinutes":null,"primary":null,"me":null,"$clusterTime":null}},"stale":false,"compatible":true,"heartbeatFrequencyMS":500,"localThresholdMS":15,"setName":null,"maxElectionId":null,"maxSetVersion":null,"commonWireVersion":0,"logicalSessionTimeoutMinutes":null
}
```

Reply:
Based on the function name the only task it should be doing is `stringify with max len`, it seems unexpected that it also provides a default value. Maybe you could set the default value before passing it to stringifyWithMaxLen. Additionally, as you mentioned, the selector can be either a string, function, or object, therefore each of them might have a different default value rather than `custom function`. How later selectors are being used? Are they parsed in any way? It would be nice to double-check and ensure that the generic plain string value won't break any existing parsing logic if it is expect to be an object-like value.

Reply:
Yes that makes sense, I've changed it to use the `Function.toString() ` method Selectors as strings as turned back into objects if possible by unified spec runner when its running.